salt/utils/templates.py

```diff
@@ -99,6 +97,54 @@ def __getattr__(self, name):
 
 
 def wrap_tmpl_func(render_str):
+    def _sls_context(tmplpath, sls):
```

Comment:
No need to make this a local function. That just makes it harder to test, and additionally has to re-create the function every time `wrap_tmpl_func` is called.

Reply:
Ok, any issue with making it a module function. I stuck it here for now because I hated the large block of self contained code embedded, but was not sure where is the right place for it.

Reply:
Moved it to module level